WordPress/Classes/Models/AbstractPost.swift

```diff
@@ -95,4 +95,23 @@ extension AbstractPost {
             return nil
         }
     }
+
+    // MARK: - Updating the Remote Status
+    /// Updates the post after an upload failure.
+    ///
+    /// - Important: This logic could have been placed in the setter for `remoteStatus`, but it's my belief
+    ///     that our code will be much more resilient if we decouple the act of setting the `remoteStatus` value
+    ///     and the logic behind processing an upload failure.  In fact I think the `remoteStatus` setter should
+    ///     eventually be made private.
+    ///
+    @objc
+    func failedToUpload() {
```

Comment:
A tiny concern for this is that the `change()` method, which calls `failedToUpload()`, gets called every time a **media** upload fails. It's an unnecessary data operation. 

https://github.com/wordpress-mobile/WordPress-iOS/blob/9d67d09955aa5aa2243fdffc50b7490e23b3041e/WordPress/Classes/Services/PostCoordinator.swift#L83-L84

So the `dateModified` gets updated every time. I wonder if we should add a `guard`? 🤔 

```swift
guard remoteStatus != .failed, !hasRemote() else {
    return
}
```

Reply:
Done in `PostService+markAsFailedAndDraft.swift`. :)